lib/index.js

```diff
@@ -17,50 +18,61 @@
   }
 
   function isOriginAllowed(origin, allowedOrigin) {
+    debug('test of origin %s. AllowedOrigin: %O',  origin, allowedOrigin);
     if (Array.isArray(allowedOrigin)) {
       for (var i = 0; i < allowedOrigin.length; ++i) {
         if (isOriginAllowed(origin, allowedOrigin[i])) {
+          debug('isAllowedOrigin true');
           return true;
         }
       }
       return false;
     } else if (isString(allowedOrigin)) {
+      debug('isAllowedOrigin %s', origin === allowedOrigin);
       return origin === allowedOrigin;
     } else if (allowedOrigin instanceof RegExp) {
+      debug('isAllowedOrigin %s' , allowedOrigin.test(origin));  
       return allowedOrigin.test(origin);
     } else {
+      debug('isAllowedOrigin %s',!!allowedOrigin)
       return !!allowedOrigin;
     }
   }
 
   function configureOrigin(options, req) {
+    debug('Configure Origin');
     var requestOrigin = req.headers.origin,
       headers = [],
       isAllowed;
 
     if (!options.origin || options.origin === '*') {
       // allow any origin
+      debug('added header Access-Control-Allow-Origin *');
```

Comment:
Rather than duplicating this on every single header add line, you could just place a single call when the headers are populated to list them.